src/main/java/com/microsoft/cse/helium/app/services/keyvault/KeyVaultService.java

```diff
@@ -0,0 +1,104 @@
+
+    public KeyVaultService (@Value("${helium.keyvault.name}") String keyVaultName, @Value("${helium.environment.flag}") String environmentFlag) throws Exception {
+        _keyVaultName = keyVaultName.trim().toUpperCase();
+        _environmentFlag = environmentFlag.trim().toUpperCase();
+
+        // validate params and create credential object
+        if(_environmentFlag.equals("MSI")) {
+            _credentials = new MSICredentials(AzureEnvironment.AZURE);
+        } else if (_environmentFlag.equals("CLI")) {
+            try{
+                _credentials = AzureCliCredentials.create();
+            }
+            catch (IOException ex)
+            {
+                _logger.error(ex.getMessage());
+                throw(ex);
+            }
+        } else {
+            throw new IllegalArgumentException("helium.environment.flag (value='" + _environmentFlag + "') must be 'MSI' or 'CLI'.  Check ${He_EnvironmentFlag} in your environment variables.");
+        }
+        
+        if(!checkKeyVaultName(_keyVaultName)) {
+            throw new IllegalArgumentException("helium.keyvault.name (value='" + _keyVaultName + "') must be 3-24 characters long, begin with a character, may contain alphanumeric or hyphen, no repeating hyphens, and end with alphanumeric.  Check ${KeyVaultName} in your environment variables.");
+        }
+
+        _keyVaultClient = new KeyVaultClient(_credentials);
+
+    }
+
+    private Boolean checkKeyVaultName(String keyVaultName){
+        Boolean validSetting = true;
+
+        if(keyVaultName.length() < 3 || keyVaultName.length() > 24){
+            validSetting= false;
+        }
+
+        // validate key vault name with regex: ^[a-zA-Z](?!.*--)([a-zA-Z0-9-]*[a-zA-Z0-9])?$
+        //    ^[a-zA-Z]       - start of string, must be a alpha character
+        //    (?!.*--)        - look ahead and make sure there are no double hyphens (e.g., "--")
+        //    [a-zA-Z0-9-]*   - match alphanumeric and hyphen as many times as needed
+        //    [a-zA-Z0-9]     - final character must be alphanumeric
+        if(!keyVaultName.matches((_keyVaultNameRegex))) {
+            validSetting=false;
+        }
+
+        return validSetting;
+    }
+
+    public String getKeyVaultUri(){
+        String kvUri = "https://" + _keyVaultName + ".vault.azure.net";
+        return kvUri;
+    }
+
+    public String getSecret(String secretName){
+        String returnValue="";
+        
+        String kvUri = getKeyVaultUri();
+
+        SecretBundle secret = _keyVaultClient.getSecret(kvUri, secretName);
+        returnValue = secret.value();
+
+        return returnValue;
+    }
+
+    public KeyBundle getKey (String keyName){
+
+        return _keyVaultClient.getKey(getKeyVaultUri(), keyName);
+    }
+
+    public CertificateBundle getCertificate (String certName){
+
+        return _keyVaultClient.getCertificate(getKeyVaultUri(), certName);
+
```

Comment:
inconsistent use of space after return. Here it is not needed.

Reply:
Removed.